src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -234,120 +249,116 @@ public void update() {
-                if (fuelRemaining > 0)
-                    fuelRemaining--;
+
 
             }
+        }
+        if (fuelRemaining > 0)
+            fuelRemaining--;
 
-            if (inventory.getStackInSlot(0) != null && inventory.getStackInSlot(1) == null) {
-                ItemStack itemIn = inventory.getStackInSlot(0);
-                ItemStack itemOut;
 
-                if (!PulverizerRegistry.containsInput(itemIn))
-                    return;
+        //code for item in
+        if (inventory.getStackInSlot(0) != null && inventory.getStackInSlot(1) == null) {
+            ItemStack itemIn = inventory.getStackInSlot(0);
+            ItemStack itemOut;
 
-                if (itemIn.stackSize - 1 <= 0) {
-                    itemOut = itemIn.copy();
-                    itemIn = null;
-                } else {
-                    itemOut = itemIn.copy();
+            if (!PulverizerRegistry.containsInput(itemIn))
+                return;
 
-                    itemOut.stackSize = 1;
-                    itemIn.stackSize = itemIn.stackSize - 1;
-                }
+            if (itemIn.stackSize - 1 <= 0) {
+                itemOut = itemIn.copy();
+                itemIn = null;
+            } else {
+                itemOut = itemIn.copy();
 
-                if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
-                if (itemOut.stackSize == 0) itemOut = null;
+                itemOut.stackSize = 1;
+                itemIn.stackSize = itemIn.stackSize - 1;
+            }
 
-                inventory.setInventorySlotContents(0, itemIn);
-                inventory.setInventorySlotContents(1, itemOut);
+            if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
+            if (itemOut.stackSize == 0) itemOut = null;
 
-                ticksRemaining = 200;
-                machineActive = true;
+            inventory.setInventorySlotContents(0, itemIn);
+            inventory.setInventorySlotContents(1, itemOut);
 
-                this.markForUpdate();
-                this.markDirty();
-            }
+            ticksRemaining = 200;
+            machineActive = true;
 
+            this.markForUpdate();
+            this.markDirty();
+        }
 
-            if (fuelRemaining == 0 && machineActive && machineTier == machineTier.TIER_0) {
-                ticksRemaining = 0;
-                test = false;
 
-            } else {
-                test = false;
-            }
+        if (ticksRemaining <= 0 && machineActive) {
 
 
-            if (ticksRemaining <= 0 && machineActive) {
+            ticksRemaining = 0;
 
-                if (test == false) {
-                    System.out.println(test);
-                    ticksRemaining = 0;
+            if (worldObj.isRemote)
+                return;
 
-                    if (worldObj.isRemote)
-                        return;
+            ItemStack processItem = inventory.getStackInSlot(1);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS134)